Core/Core/Quizzes/QuizDetails/ViewModel/QuizSubmissionBreakdownViewModel.swift.swift

```diff
@@ -0,0 +1,87 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2022-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public class QuizSubmissionBreakdownViewModel: SubmissionBreakdownViewModelProtocol {
+
+    @Published public var isReady: Bool = false
+    @Published public var showError: Bool = false
+    @Published public private(set) var errorText: String?
+
+    public var graded: Int = 0
+    public var ungraded: Int = 0
+    public var unsubmitted: Int = 0
+    public var submissionCount: Int = 0
```

Comment:
Since these properties are accessed by the UI and can change in response to CoreData modifications I would add `@Published`.